src/plugins/ucx/ucx_utils.cpp

```
@@ -103,57 +103,61 @@ nixlUcxEp::err_cb(ucp_ep_h ucp_ep, ucs_status_t status) {
-    switch (state) {
+    switch (current_state) {
     case nixl::ucx::ep_state_t::UNINITIALIZED:
     case nixl::ucx::ep_state_t::DISCONNECTED:
         // The EP has not been connected, or already disconnected.
         // Nothing to do.
         NIXL_ASSERT(eph == nullptr);
         return NIXL_SUCCESS;
     case nixl::ucx::ep_state_t::FAILED:
-        // The EP was closed in error callback, just return error.
+        request = ucp_ep_close_nb(eph, UCP_EP_CLOSE_MODE_FORCE);
+        if (UCS_PTR_IS_PTR(request)) {
```

Cppcheck (2.21.0)

[style] The function 'getBase' is never used.

(unusedFunction)